lang/transformations/src/erasure/symbol_table.rs

```diff
@@ -0,0 +1,317 @@
+    pub modules: HashMap<Url, ModuleErasureSymbolTable>,
+}
+
+impl GlobalErasureSymbolTable {
+    pub fn insert(&mut self, uri: Url, table: ModuleErasureSymbolTable) {
+        self.modules.insert(uri, table);
+    }
+
+    pub fn lookup_data(&self, uri: &Url, name: &str) -> Result<&Data, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_data
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown data type: {}", name)))
+    }
+
+    pub fn lookup_codata(&self, uri: &Url, name: &str) -> Result<&Codata, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_codata
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown codata type: {}", name)))
+    }
+
+    pub fn lookup_def(&self, uri: &Url, name: &str) -> Result<&DefMeta, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_def
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown definition: {}", name)))
+    }
+
+    pub fn lookup_codef(&self, uri: &Url, name: &str) -> Result<&CodefMeta, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_codef
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown codefinition: {}", name)))
+    }
+
+    pub fn lookup_let(&self, uri: &Url, name: &str) -> Result<&LetMeta, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_let
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown let binding: {}", name)))
+    }
+
+    pub fn lookup_ctor(&self, uri: &Url, name: &str) -> Result<&Ctor, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_ctor
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown constructor: {}", name)))
+    }
+
+    pub fn lookup_dtor(&self, uri: &Url, name: &str) -> Result<&Dtor, ErasureError> {
+        let module = self.lookup_module(uri)?;
+        module
+            .map_dtor
+            .get(name)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown destructor: {}", name)))
+    }
+
+    fn lookup_module(&self, uri: &Url) -> Result<&ModuleErasureSymbolTable, ErasureError> {
+        self.modules
+            .get(uri)
+            .ok_or_else(|| ErasureError::Impossible(format!("Unknown module: {}", uri.as_str())))
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct ModuleErasureSymbolTable {
+    pub map_data: HashMap<String, Data>,
+    pub map_codata: HashMap<String, Codata>,
+    pub map_def: HashMap<String, DefMeta>,
+    pub map_codef: HashMap<String, CodefMeta>,
+    pub map_let: HashMap<String, LetMeta>,
+    pub map_ctor: HashMap<String, Ctor>,
+    pub map_dtor: HashMap<String, Dtor>,
+}
+
```

Comment:
The table entries can be trimmed down to collections of `erased_params: Vec<bool>`s.